src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -56,6 +60,11 @@ public void resetData(ReadOnlyAddressBook newData) {
         requireNonNull(newData);
 
         setPersons(newData.getPersonList());
+        setMedicationMap(newData.getMedicationMapObject());
+    }
+
+    private void setMedicationMap(String medicationMap) {
+        this.medicationMap.setMedicationMap(medicationMap);
```

Comment:
This might be confused with recursion on first glance, but I understand the two methods are from different classes. Perhaps renaming `setMedicationMap` in
```java
 this.medicationMap.setMedicationMap(medicationMap);
```
would be a better option.

Reply:
Fair enough. I've renamed them to imply the usage of stringified versions of the map.